File: sub-web-service/src/main/java/org/github/web/system/service/IShiroService.java
package org.github.web.system.service;

import lombok.*;
import org.github.web.model.shiro.AuthorInfo;
import org.github.web.model.shiro.UserDTO;

import java.util.List;

public interface IShiroService {
  @NonNull List<AuthorInfo> queryAuthorInfo(@NonNull Long userId);

  @NonNull UserDTO queryUser(@NonNull String username);
}

File: sub-mysql-mydb-service/src/main/java/org/github/mysql/mydb/auth/service/IShiroService.kt
package org.github.mysql.mydb.auth.service

import org.github.shiro.AuthorInfo
import org.github.shiro.User
import org.github.spring.restful.json.JSONArrayReturn
import org.github.spring.restful.json.JSONDataReturn

interface IShiroService {
  fun queryAuthorInfo(userId: String): JSONArrayReturn<AuthorInfo>

  fun queryUser(username: String): JSONDataReturn<out User>
}
